### PR TITLE
Update octicons auto-update & metadata

### LINKED ISSUE #12851: Octicon is not automatically updated
While on [github](https://github.com/primer/octicons/releases) and [npm](https://www.npmjs.com/package/octicons) oction is in version 7.3.0 on [CDNJS](https://cdnjs.com/libraries/octicons) it's still on 4.4.0.

CDNJS says that automatic updates are enabled for octicon but it seems like this update didn't happen recently. Maybe a bug? Maybe the metadata wrong?

### PR DESCRIPTION
**Pull request by sashberd**
Update homepage property to https
Update auto update config from git to npm
close #12851

Pull request for issue: #12851

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars; clean, clear and easy to understand.
 * [x] The parent of the commit(s) in the PR is not older than 3 days.
 * [x] Pull request is sent from a non-master branch with a meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/octicons/package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "OFL-1.1",
```
Comment (extend1994) [inline ajax/libs/octicons/package.json]: Please help update license, only `MIT` is available now, this change can be with `homepage` update.
Ref: https://github.com/primer/octicons/blob/master/LICENSE

File: ajax/libs/octicons/package.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
...
     "MIT"
   ],
-  "homepage": "http://octicons.github.com/",
+  "homepage": "https://octicons.github.com/",
```
Comment (extend1994) [inline ajax/libs/octicons/package.json]: Please separate this update and auto-update change into different commits.

File: ajax/libs/octicons/package.json [hunk 3]
```diff
@@ -17,18 +17,15 @@
...
-        ]
-      }
-    ]
-  },
+  "npmName": "octicons",
+  "npmFileMap": [
+    {
+      "basePath": "build",
+      "files": [
+        "/**/!(*.json|*.html)"
```
Comment (extend1994) [inline ajax/libs/octicons/package.json]: `**/!(*.json|*.html)`, please run git/npm auto-update test before you sent the PR.

Comment (sashberd) [reply ajax/libs/octicons/package.json]: I did this command and this pattern worked. What is problem with it?

Comment (extend1994) [reply ajax/libs/octicons/package.json]:
```
events.js:183
      throw er; // Unhandled 'error' event
      ^

Error: unexpected end of file
    at Gunzip.zlibOnError (zlib.js:153:15)
```

It also could be my computer's problem, but I suppose relative path `**/!(*.json|*.html)` makes more sense.

**Comment by sashberd** [comment]
@extend1994 The PR was updated. FYR

File: ajax/libs/octicons/package.json [hunk 4]
```diff
@@ -4,10 +4,9 @@
   "version": "4.4.0",
   "description": "GitHub's icon font",
   "licenses": [
-    "OFL-1.1",
     "MIT"
```
Comment (extend1994) [inline ajax/libs/octicons/package.json]: Only one license left, so
```js
  "license": "MIT",
```

should be used.

**Pull request merged**